donate/utility/acoustic/acoustic.py

```diff
@@ -116,7 +120,28 @@ def _call_xt(self, xml):
     def send_mail(self, to, campaign_id, fields=None, bcc=None, save_to_db=False):
         # If we are testing, do not send emails
         if not settings.TESTING:
-            self._call_xt(transact_xml(to, campaign_id, fields, bcc, save_to_db))
+            send_retries = 3
+
+            for attempt in range(send_retries):
+
+                if attempt == 0:
+                    # If this is the first try, send the email immediately.
+                    schedule.enter(0, 1, self._call_xt, (to, campaign_id, fields, bcc, save_to_db))
+                else:
+                    # If first try is unsuccessful, wait 10 seconds and try again.
+                    schedule.enter(10, 1, self._call_xt, (to, campaign_id, fields, bcc, save_to_db))
```

Comment:
Can we explain why we're only waiting 10 seconds? Is there an issue where we agreed on that time, or is there some information that future readers can use to understand why we're using this specific value?

On that note, let's remove some duplication, and make sure we don't have "magic constants" in the code:

```python
    def send_mail(self, to, campaign_id, fields=None, bcc=None, save_to_db=False):
        if settings.TESTING:
            return

        send_mail_immediately = 0  # seconds
        send_mail_delay = 10  # seconds
        schedule_priority = 1
        call_args = (to, campaign_id, fields, bcc, save_to_db)

        for attempt in range(send_retries):
            if attempt == 0:
                schedule.enter(send_mail_immediately, schedule_priority, self.attempt_send_mail, call_args)
            else:
                schedule.enter(send_mail_delay, schedule_priority, self.attempt_send_mail, call_args)

            try:
                schedule.run(blocking=True)  # explicitly block, just in case the implicit behaviour changes in the future
                break
            except ConnectionError:
                ...
```

Reply:
Hi @Pomax, no explicit reason for the 10 seconds, I just added it as I felt having the user wait more than 30 seconds might not be the best experience, but will ping @jzinner to see what they think about the time and whether they would like to change it!

Reply:
Does this logic in acoustic.py run in real-time when the user clicks submit?  

@danielfmiranda when you test this in development with 10 second retries, are you experiencing a 30 second wait wait for the retries to finish - until the thank you message appears?  

If this runs in real-time and the user has to wait for the retries then 10 seconds is too long and wait 3 seconds between retries.
If this runs in the background and the user does not have to wait for this to finish, then 10 seconds is ok.

Reply:
Hi @jzinner! 

The thank you message appears as soon as the payment goes through on the braintree side. So this would operate in the second option, which is the user receives the thank you message and the message will try to send again in 10 seconds in the background. 

In that case, I will leave it at 10 seconds. Thank you!